plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/AbstractStringProcessor.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.ingest.processor;
+
+import org.elasticsearch.ingest.Data;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public abstract class AbstractStringProcessor implements Processor {
```

Comment:
I think we can always change it when we need to. In general, trying to reuse this class further will make it more complicated (same reason why I didn't do it with other string processors like split and join, because their configuration is provided in a different format and it's not worthwhile to make this class even more generic). That is why I think it's ok as-is, it is not meant to be reused besides trim, uppercase and lowercase for now, we will see what comes up next though :)